core/http/routes/welcome.go

```diff
@@ -0,0 +1,42 @@
+package routes
+
+import (
+	"github.com/go-skynet/LocalAI/core/config"
+	"github.com/go-skynet/LocalAI/internal"
+	"github.com/go-skynet/LocalAI/pkg/model"
+	"github.com/gofiber/fiber/v2"
+)
+
+func RegisterWelcomeRoute(
+	app *fiber.App,
+	cl *config.BackendConfigLoader,
+	ml *model.ModelLoader,
+	appConfig *config.ApplicationConfig,
+	auth func(*fiber.Ctx) error,
+) {
+	if appConfig.DisableWelcomePage {
+		return
+	}
+
+	models, _ := ml.ListModels()
+	backendConfigs := cl.GetAllBackendConfigs()
+
+	app.Get("/", auth, func(c *fiber.Ctx) error {
```

Comment:
Style Question (because I'm honestly not sure which I prefer yet)

Should the actual functionality be here, under `core/http/routes`? Or perhaps the function _body_ should live under `core/http/endpoints/localai/index.go`? 

At the end of the day, I think having both `routes` and `endpoints` is a bit... confusing and I'd like to settle on one or the other. If we do delete this folder entirely, we could move `core/http/routes/api.go` to `core/http/routes.go` in order to keep it separate from `app.go` which I really like!

Reply:
good question, up to now I just did this distinction mentally: 

- `routes`: where we register the endpoints with the handlers
- `endpoints`: where the handlers logic lays on (especially if have big logic corpus)

maybe makes sense to move the handler in `endpoints` as well to be consistent for now, we can still decide to rename that later on

Reply:
slightly adapted, now the logic is in `endpoints`!